contracts/debtAllocators/YieldManager/YieldManager.sol

```diff
@@ -0,0 +1,388 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.8.18;
+
+import {IVault} from "@yearn-vaults/interfaces/IVault.sol";
+import {AprOracle} from "@periphery/AprOracle/AprOracle.sol";
+
+import {Keeper, Governance} from "./Keeper.sol";
+import {GenericDebtAllocator} from "../GenericDebtAllocator.sol";
+
+/**
+ * @title YearnV3 Yield Yield Based Debt Allocator
+ * @author yearn.finance
+ * @notice
+ *  This Debt Allocator is meant to be used alongside
+ *  a Yearn V3 vault to allocate funds to the optimal strategy.
+ */
+contract YieldManager is Governance {
+    /// @notice Emitted when the open flag is updated.
+    event UpdateOpen(bool status);
+
+    /// @notice Emitted when a proposer status is updated.
+    event UpdateProposer(address indexed proposer, bool status);
+
+    /// @notice Emitted when a vaults status is updated.
+    event UpdateVaultAllocator(address indexed vault, address allocator);
+
+    // Struct that contains the address of the strategy and its best allocation.
+    struct Allocation {
+        // Address of the strategy.
+        address strategy;
+        // Debt for the strategy to end with.
+        // Can take 79 Billion 18 decimal tokens.
+        uint96 newDebt;
+    }
+
+    /// @notice Only allow the sender to be an allocator if not opened.
+    modifier onlyProposersOrOpen() {
+        _isProposerOrOpen();
+        _;
+    }
+
+    /// @notice Check if it has been opened or is an allocator.
+    function _isProposerOrOpen() internal view {
+        require(proposer[msg.sender] || open, "!allocator or open");
+    }
+
+    uint256 internal constant MAX_BPS = 10_000;
+
+    /// @notice Contract that holds the logic and oracles for each strategy.
+    AprOracle internal constant aprOracle =
+        AprOracle(0xF012fBb9283e03994A7829fCE994a105cC066c14);
```

Comment:
Any reason this is a hardcoded address?

Reply:
Its one of the few contracts i am confident in that wont need to change.